### PR TITLE
[lutron] Add support for GRAFIK Eye in RadioRA 2/HomeWorks QS

### PR DESCRIPTION
**Pull request by bobadair**
Hi. This PR adds support for GRAFIK Eye QS front panel keypads when used in Lutron RadioRA 2 and HomeWorks QS systems. The model parameter allows for selection of the 0, 1, 2, or 3 column variations of the front panel. A related doc update is included.

It also includes a fix for a bug that caused an exception when re-initializing any keypad thing after modifying the "model" parameter. This consists of a small change to BaseKeypadHandler.

Finally, it adds me to the CODEOWNERS file line for the Lutron binding.  An increasing amount of the code in the binding is mine, so I'd like to be notified if there is a problem with it.  I hope I'm not stepping on any toes here.

Regards,
Bob

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -80,7 +80,7 @@
 /bundles/org.openhab.binding.lirc/ @kabili207
 /bundles/org.openhab.binding.logreader/ @paulianttila
 /bundles/org.openhab.binding.loxone/ @ppieczul
-/bundles/org.openhab.binding.lutron/ @actong
+/bundles/org.openhab.binding.lutron/ @actong @bobadair
```
Comment (J-N-K) [inline CODEOWNERS]: I think - compared to the original contribution - this PR adds only a small amount of code. Please leave the original line in place.

File: bundles/org.openhab.binding.lutron/README.md [hunk 2]
```diff
@@ -185,11 +195,14 @@ The correct channels will then be automatically configured.
 
 To support this keypad's contact closure inputs, CCI channels named *cci1* and *cci2* are created with item type Contact and category Switch.
 They are marked as Advanced, so they will not be automatically linked to items in the Paper UI's Simple Mode.
-They accept ON/OFF commands and present ON/OFF states the same as a keypad button.
+They present ON/OFF states the same as a keypad button.
```
Comment (J-N-K) [inline bundles/org.openhab.binding.lutron/README.md]: This is a difference. Was the "accept ON/OFF" part wrong?

File: bundles/org.openhab.binding.lutron/README.md [hunk 3]
```diff
@@ -289,7 +331,7 @@ Therefore, the channel state will only transition OFF->ON->OFF when you send a O
 
 For maintained CCOs, sending ON and OFF commands works as expected, and the channel state updates as expected when either openHAB commands or external events change the CCO device state.
 
-Example:
+Thing configuration file example:
```
Comment (J-N-K) [inline bundles/org.openhab.binding.lutron/README.md]: Why did you change all these?

File: bundles/org.openhab.binding.lutron/README.md [hunk 4]
```diff
@@ -337,16 +379,16 @@ A setting of 0 will disabled polling.
...
 
 ```
 Thing greenmode greenmode [ integrationId=22 ]
 ```
 
 ### Timeclock
 
-Radio RA2 and Homeworks QS have timeclock subsystems that provide scheduled execution of tasks at set times, randomized times, or at arbitrary offsets from local sunrise/sunset.
-The tasks executed depend on which timeclock mode (e.g. Normal, Away, Suspend) is currently selected, and the modes themselves are user-definable (Radio RA2 only).
+RadioRA 2 and Homeworks QS have timeclock subsystems that provide scheduled execution of tasks at set times, randomized times, or at arbitrary offsets from local sunrise/sunset.
```
Comment (J-N-K) [inline bundles/org.openhab.binding.lutron/README.md]:
```suggestion
RadioRA 2 and Homeworks QS have timeclock subsystems that provide scheduled execution of tasks at set times, randomized times or at arbitrary offsets from local sunrise/sunset.
```

File: bundles/org.openhab.binding.lutron/README.md [hunk 5]
```diff
@@ -337,16 +379,16 @@ A setting of 0 will disabled polling.
...
+The tasks executed depend on which timeclock mode (e.g. Normal, Away, Suspend) is currently selected, and the modes themselves are user-definable (RadioRA 2 only).
```
Comment (J-N-K) [inline bundles/org.openhab.binding.lutron/README.md]:
```suggestion
The tasks executed depend on the currently selected timeclock mode (e.g. Normal, Away, Suspend) and the modes themselves are user-definable (RadioRA 2 only).
```

File: bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/discovery/project/DeviceType.java [hunk 6]
```diff
@@ -26,5 +27,5 @@
     QS_IO_INTERFACE,
     SEETOUCH_KEYPAD,
     SEETOUCH_TABLETOP_KEYPAD,
-    VISOR_CONTROL_RECEIVER
+    VISOR_CONTROL_RECEIVER,
```
Comment (J-N-K) [inline bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/discovery/project/DeviceType.java]: Please revert

File: bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/GrafikEyeKeypadHandler.java [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+ * a RadioRA 2 or HomeWorks QS System.
+ *
+ * Does not communicate with the scene controller, timeclock controller, or wireless
+ * and EcoSystem occupancy sensors.
+ *
+ * @author Bob Adair - Initial contribution
+ */
+public class GrafikEyeKeypadHandler extends BaseKeypadHandler {
+
+    private static enum Component implements KeypadComponent {
```
Comment (J-N-K) [inline bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/GrafikEyeKeypadHandler.java]: Can you move that to a another file?

File: CODEOWNERS (hunk 1 above)
Comment (bobadair) [reply CODEOWNERS]: Yes, I agree there isn't much in this commit. :-) It's the 12 that went before that I was really referring to.  I think I'm really the only one working on this binding in the past year or so.  I'll change it back if you think I should, though.  I don't want to upset anyone.

File: bundles/org.openhab.binding.lutron/README.md (hunk 2 above)
Comment (bobadair) [reply bundles/org.openhab.binding.lutron/README.md]: Yes. That's fixing my own error from a previous commit.

File: bundles/org.openhab.binding.lutron/README.md (hunk 3 above)
Comment (bobadair) [reply bundles/org.openhab.binding.lutron/README.md]: Just to clarify that those examples only include excerpts from thing files.  I had updated all of the examples in a previous commit and forgot to make that change.  So I sort of snuck it in here. The README has been a work in progress.

File: bundles/org.openhab.binding.lutron/README.md (hunk 5 above)
Comment (bobadair) [reply bundles/org.openhab.binding.lutron/README.md]: Yes, that sounds better.  I'll change it.

File: bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/GrafikEyeKeypadHandler.java (hunk 7 above)
Comment (bobadair) [reply bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/GrafikEyeKeypadHandler.java]: I could, but it will only ever be used by GrafikEyeKeypadHandler. Every keypad handler (or CCI handler) defines a different Component enum which encodes the component numbering for that particular series of keypads.  Since there are 8 of them, with another 4-5 still to be created, it seemed cleaner to define them all inside the classes they are used by.

**Comment by bobadair** [comment]
Thanks @J-N-K . I've pushed a commit containing some changes.  Please let me know if there are any other changes that you think should be made. I'd like to wrap this one up and make sure it doesn't cause any merge conflicts for #5874, which is more complicated.

File: CODEOWNERS (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaikreuzer) [reply CODEOWNERS]: Done!

[the remark this answers is not in the document]
Comment (bobadair) [reply CODEOWNERS]: Thanks, @cweitkamp and @kaikreuzer!
